eclipse-collections-code-generator/src/main/resources/impl/list/immutable/immutablePrimitiveSingletonList.stg

```diff
@@ -439,6 +444,50 @@ final class Immutable<name>SingletonList implements Immutable<name>List, Seriali
         }
         return Lists.immutable.with(PrimitiveTuples.pair(this.element1, Iterate.getFirst(iterable)));
     }
+    <if(primitive.specializedStream)>
+        @Override
+        public Spliterator.Of<name> primitiveSpliterator(){
+           return new SingletonSpliterator(this.element1);
+        }
+        static private class SingletonSpliterator implements Spliterator.Of<name>{
```

Comment:
Can you extract this out into its own public class? It'll be useful for doing the same thing with other singleton collections (e.g. sets).

Reply:
Done